pyetrade/order.py

```diff
@@ -10,13 +10,68 @@
 
 """
 
+import dateutil.parser
 import logging
 import jxmlease
+from nose.tools import *
+import pdb
 from requests_oauthlib import OAuth1Session
 
 LOGGER = logging.getLogger(__name__)
 
 
+# some constants
+CALL = "Call"
+PUT  = "Put"
+
+
+# resp_format: xml (default) or json
+# empty_json: either [] or {}, depends on the caller's semantics
+def get_request_result(req, resp_format, empty_json):
+    assert resp_format in ("json", "xml", None)  # TODO: why None?
+
+    LOGGER.debug(req.text)
+    req.raise_for_status()
+
+    if resp_format == "json":
+      if req.text.strip() == "":
+        # otherwise, when ETrade server return empty string, we got this error:
+        # simplejson.errors.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
+        return empty_json  # empty json object
+      else:
+        return req.json()
+
+    if resp_format is None:  # TODO(jessecooper): should this be: == "xml"?
+        return jxmlease.parse(req.text)
+
+    return req.text
+
+
+# return Etrade internal option symbol: e.g. "PLTR--220218P00023000" ref:_test_option_symbol()
+def option_symbol(symbol, callPut, expiryDate, strikePrice):
+  sym = symbol.strip().upper()
+  symstr = sym + ("-" * (6 - len(sym)))
+
+  ed = dateutil.parser.parse(expiryDate)  # dateutil can handle most date formats
+  edstr = ed.strftime("%y%m%d")
+  assert_equal(len(edstr), 6)
+
+  sp = "%08d" % (float(strikePrice) * 1000)
+  assert_equal(len(sp), 8)
+
+  opt_sym = symstr + edstr + callPut.strip().upper()[0] + sp
+  assert_equal(len(opt_sym), 21)
+  return opt_sym
+
+
+def _test_option_symbol():

```

Comment:
Can we move this to the test method in the test file. The goal of this is to be able to remove test frameworks from production installs.

Reply:
Moved.

I'm used to unittest in DLang:

https://tour.dlang.org/tour/en/gems/unittesting

where the convention is to write `unittest` as closer to the function / class being tested as possible, so:
-- it's easier to maintain than in two separate files, e.g. after code change.
-- also serve as code example of how to use the function / class.

The D compiler will ignore the `unittest` section during release build.